process/block/preprocess/miniBlocksCompaction.go

```diff
@@ -146,6 +132,20 @@ func (mbc *miniBlocksCompaction) calculateUsedGasInMiniblock(miniBlock *block.Mi
 	return gasUsedInMiniBlock, nil
 }
 
+func (mbc *miniBlocksCompaction) computeGasConsumedByTx(txHash []byte) (uint64, error) {
+	tx, ok := mbc.mapHashToTx[string(txHash)]
+	if !ok {
+		return 0, process.ErrMissingTransaction
+	}
+
+	txGasLimit := mbc.economicsFee.ComputeGasLimit(tx)
+	if isSmartContractAddress(tx.GetRecvAddress()) {
```

Comment:
this should be done only if destination is not current shard.

Reply:
you can leave it right now. As this restriction is what we need. Actual gas consumption will be added on a new pull request.

Reply:
ok